tasks/show_linters_issues/golangci_lint_parser.py

```diff
@@ -0,0 +1,110 @@
+"""
+This module defines a parser for golangci-lint output.
+"""
+
+import re
+import os
+from collections import defaultdict
+from typing import Dict
+from ..libs.pipeline_notifications import read_owners
+
+# Example lint message
+# "pointer.go:6:1: package-comments: should have a package comment (revive)"
+LINT_PATTERN = re.compile("^([^:]+):([0-9]+):([0-9]+): (([^:]+): )?(.+) \\((.+)\\)$")
+
+# Example module message
+# "Linters for module /Users/pierre.gimalac/go/src/github.com/DataDog/datadog-agent/pkg/remoteconfig/state failed (base flavor)"
+MODULE_PATTERN = re.compile("^.*Linters for module ([^ ]+) failed \\((.+)\\).*$")
+
+CODEOWNERS_FILE_PATH = ".github/CODEOWNERS"
+
+def is_team_owner(file: str, team: str) -> bool:
+    codeowners = read_owners(CODEOWNERS_FILE_PATH)
+    team = team.lower()
+    file_owners = codeowners.of(file)
+    file_owners = list(map(lambda x: (x[0], x[1].lower()), file_owners))
+    return ('TEAM', team) in file_owners
+
+
+def parse_file(golangci_lint_output: str):
+    """
+    Parses the output of the golangci-lint run.
+    Returns a Dict(linter: List(base_path, row, col, lint, descr)).
+    """
+    lints = []
+    current_module = None
+    for line in golangci_lint_output.split("\n"):
+        line = line.strip()
+
+        match = re.match(MODULE_PATTERN, line)
+        if match:
+            module_path, flavor = match.groups()
+            current_module = module_path
+            continue
+
+        if current_module is None:
+            continue
+
+        match = re.match(LINT_PATTERN, line)
+        if match is None:
+            continue
+
+        file_path, row, col, lint, _, descr, linter = match.groups()
+        full_path = os.path.normpath(os.path.join(current_module, file_path))
+        base_path = full_path.removeprefix(os.getcwd()).removeprefix('/')
+        lints.append((base_path, row, col, lint, descr, linter))
+
+    lints_per_linter = dict()
```

Comment:
```suggestion
    lints_per_linter = defaultdict(list)
```
and you can remove lines 59, 60

Reply:
Missed this one thanks ! Fixed with [a78358f](https://github.com/DataDog/datadog-agent/pull/19015/commits/a78358f07bb5b8fa55810f75e060eb1dfb5a8eb9).